constructr-akka/src/multi-jvm/scala/de/heikoseeberger/constructr/akka/MultiNodeConstructrSpecConsul.scala

```diff
@@ -0,0 +1,150 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package de.heikoseeberger.constructr.akka
+
+import akka.actor.ActorDSL.{ Act, actor }
+import akka.cluster.{ Cluster, ClusterEvent }
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.client.RequestBuilding
+import akka.http.scaladsl.model.StatusCodes.{ NotFound, OK }
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.pattern.ask
+import akka.remote.testkit.{ MultiNodeConfig, MultiNodeSpec }
+import akka.stream.ActorMaterializer
+import akka.testkit.TestDuration
+import akka.util.Timeout
+import com.typesafe.config.ConfigFactory
+import de.heikoseeberger.constructr.coordination.Coordination.AddressSerialization
+import de.heikoseeberger.constructr.coordination._
+import org.scalatest.{ BeforeAndAfterAll, FreeSpecLike, Matchers }
+import scala.concurrent.Await
+import scala.concurrent.duration.DurationInt
+import scala.sys.process.{ ProcessLogger, stringToProcess }
+
+object ConstructrMultiNodeConfigConsul extends MultiNodeConfig {
+
+  val host = "docker-machine ip default".!!.trim
+
+  val nodes = 1.to(5).to[List].map(n => node(2550 + n))
+
+  private def node(port: Int) = {
+    commonConfig(ConfigFactory.load())
+    val node = role(port.toString)
+    nodeConfig(node)(ConfigFactory.parseString(
+      s"""|akka.actor.provider               = akka.cluster.ClusterActorRefProvider
+          |akka.loggers                      = ["de.heikoseeberger.akkalog4j.Log4jLogger"]
+          |akka.loglevel                     = "DEBUG"
+          |akka.remote.netty.tcp.hostname    = "127.0.0.1"
+          |akka.remote.netty.tcp.port        = $port
+          |constructr.akka.coordination.backend = "consul"
+          |constructr.akka.coordination.port = 8500
+          |constructr.akka.coordination.host = $host""".stripMargin
+    ))
+    node
+  }
+}
+
+class MultiNodeConstructrSpecConsulMultiJvmNode1 extends MultiNodeConstructrSpecConsul
+class MultiNodeConstructrSpecConsulMultiJvmNode2 extends MultiNodeConstructrSpecConsul
+class MultiNodeConstructrSpecConsulMultiJvmNode3 extends MultiNodeConstructrSpecConsul
+class MultiNodeConstructrSpecConsulMultiJvmNode4 extends MultiNodeConstructrSpecConsul
+class MultiNodeConstructrSpecConsulMultiJvmNode5 extends MultiNodeConstructrSpecConsul
+
+abstract class MultiNodeConstructrSpecConsul[A: AddressSerialization] extends MultiNodeSpec(ConstructrMultiNodeConfigConsul)
+    with FreeSpecLike with Matchers with BeforeAndAfterAll {
+  import ConstructrMultiNodeConfig._
+  import RequestBuilding._
+  import system.dispatcher
+
+  def toNodes(s: String) = {
```

Comment:
Please move to line 138, i.e. after the tests (only vals should come before the tests).